opcodes/dxil/dxil_ray_tracing.cpp

```diff
@@ -49,6 +71,14 @@ bool emit_trace_ray_instruction(Converter::Impl &impl, const llvm::CallInst *ins
 	spv::Id ray_origin_vec = impl.build_vector(builder.makeFloatType(32), ray_origin, 3);
 	spv::Id ray_dir_vec = impl.build_vector(builder.makeFloatType(32), ray_dir, 3);
 
+	auto *ray_payload = inst->getOperand(15);
+	spv::Id ray_payload_var_id = impl.get_needs_temp_storage_copy(ray_payload)
+		? emit_temp_storage_copy(impl, ray_payload, spv::StorageClassRayPayloadKHR)
+		: impl.get_id_for_value(ray_payload);
+
+	assert( impl.get_needs_temp_storage_copy(ray_payload) || impl.get_effective_storage_class(ray_payload, spv::StorageClassFunction) == spv::StorageClassRayPayloadKHR);
+	assert(!impl.get_needs_temp_storage_copy(ray_payload) || impl.get_effective_storage_class(ray_payload, spv::StorageClassFunction) == spv::StorageClassFunction);
+
```

Comment:
Also need to copy from payload to variable after the trace call is complete, since a trace call is read-write.

Reply:
Resolved locally.